docs/en/contributor/Developer-Tools.md

```diff
@@ -1,9 +1,199 @@
 ---
 layout: global
-title: Developer Tools
-nickname: Developer Tools`
+title: Contributor Tools
+nickname: Contributor Tools
 group: Contributor Resources
-priority: 0
+priority: 3
 ---
 
+* Table of Contents
+{:toc}
+
 TODO(Andrew, Zac): Migrate content from Code-Conventions, add missing content
+
+## IDE
+
+We recommend using either Eclipse or IntelliJ IDEA to contribute to Alluxio. You can generate an
+Eclipse configuration file by running:
+
+{% include Contributing-to-Alluxio/eclipse-configuration.md %}
+
+Then import the folder into Eclipse.
+
+You may also have to add the classpath variable M2_REPO by running:
+
+{% include Contributing-to-Alluxio/M2_REPO.md %}
+
+If you are using IntelliJ IDEA, you may need to change the Maven profile to 'developer' in order
+to avoid import errors. You can do this by going to
+
+    View > Tool Windows > Maven Projects
+
+## Maven Targets and Plugins
+
+Before pushing changes or submitting pull requests we recommend running various maven targets on
+your local machine to make sure your changes do not break existing behavior.
+
+For these maven commands we'll assume that your command terminal is located in the root directory
+of your locally cloned copy of the Alluxio repository.
+
+```bash
+$ cd ${ALLUXIO_HOME}
+```
+
+### Checkstyle
+
+To make sure your code follows our style conventions you may run. Note that this is run any time
+you run targets such as `compile`, `install`, or `test`.
+
+{% include Contributing-To-Alluxio/checkstyle.md %}
+
+### FindBugs
+
+Before submitting the pull-request, run the latest code against the
+[`findbugs`](http://findbugs.sourceforge.net/) Maven plugin to verify no new warnings are
+introduced.
+
+{% include Contributing-to-Alluxio/findbugs.md %}
+
+### Compilation
+
+To simply compile the code you can run. This will not execute any unit tests but will execute the
+`checkstyle` and `findbugs` plugins.
+
+```bash
+$ mvn clean compile -DskipTests
+```
+
+To speed up compilation you may try using the command:
```

Comment:
perhaps change "may try using" to "may use". It's almost certain to speed things up